examples/experimental/torch/classification/nas_advanced_main.py

```diff
@@ -0,0 +1,266 @@
+    if is_main_process():
+        configure_logging(logger, config)
+        print_args(config)
+
+    set_seed(config)
+
+    opt_config = config.get('optimizer', {})
+    # define loss function (criterion)
+    # criterion = nn.CrossEntropyLoss()
+    criterion = lambda pred, target: \
+        cross_entropy_with_label_smoothing(pred, target, opt_config.label_smoothing)
+    # criterion = criterion.to(config.device)
+
+    model_name = config['model']
+    train_criterion_fn = inception_criterion_fn if 'inception' in model_name else default_criterion_fn
+
+    nncf_config = config.nncf_config
+    pretrained = is_pretrained_model_requested(config)
+
+    # Data loading code
+    train_dataset, val_dataset = create_datasets(config)
+    train_loader, _, val_loader, _ = create_data_loaders(config, train_dataset, val_dataset)
+
+    bn_adapt_args = BNAdaptationInitArgs(data_loader=wrap_dataloader_for_init(train_loader), device=config.device)
+    nncf_config.register_extra_structs([bn_adapt_args])
+    # create model
+    model = load_model(model_name,
+                       pretrained=pretrained,
+                       num_classes=config.get('num_classes', 1000),
+                       model_params=config.get('model_params'),
+                       weights_path=config.get('weights'))
+
+    model.to(config.device)
+
+    validate(val_loader, model, criterion, config)
+
+    optimizer = get_optimizer(model, opt_config)
+
+    def train_epoch_fn(loader, model_, compression_ctrl, epoch, optimizer_):
+        train_epoch(loader, model_, criterion, train_criterion_fn, optimizer_, compression_ctrl, epoch, config,
+                    train_iters=config.train_steps, log_training_info=True)
+
+    def validate_model_fn(model_, loader):
+        top1, top5, loss = validate(loader, model_, criterion, config, log_validation_info=False)
+        return top1, top5, loss
+
+    nncf_network = create_nncf_network(model, nncf_config)
+
+    resuming_checkpoint_path = config.resuming_checkpoint_path
+    if resuming_checkpoint_path is None:
+        training_algorithm = EpochBasedTrainingAlgorithm.from_config(nncf_network, nncf_config)
+    else:
+        training_algorithm = EpochBasedTrainingAlgorithm.from_checkpoint(nncf_network, bn_adapt_args,
+                                                                         resuming_checkpoint_path)
+
+    if 'train' in config.mode:
+        # Validate supernetwork
+        top1, _, _ = validate(val_loader, model, criterion, config)
+
+        nncf_network, elasticity_ctrl = training_algorithm.run(train_epoch_fn, train_loader,
+                                                               validate_model_fn, val_loader, optimizer,
+                                                               config.checkpoint_save_dir, config.tb,
+                                                               config.train_steps)
+
+        search_algo = SearchAlgorithm(model, elasticity_ctrl, nncf_config)
```

Comment:
Propose the following API:

Create or resume search with default evaluators.
```suggestion
    if resuming_checkpoint_path is None:
        search_algo = SearchAlgorithm.from_config(model, elasticity_ctrl, nncf_config)
    else:
        search_algo = SearchAlgorithm.from_checkpoint(model, elasticity_ctrl, bn_adapt_args,
                                                                         resuming_checkpoint_path)
```


Create or resume search with external evaluators.
```python
    evaluators = [...]
    if resuming_checkpoint_path is None:
        search_algo = SearchAlgorithm.from_config(model, elasticity_ctrl, nncf_config)
    else:
        evaluators = ExternalEvaluator.from_state(path_to_external_evaluator)
        search_algo = SearchAlgorithm.from_checkpoint(model, elasticity_ctrl, bn_adapt_args,
                                                                         resuming_checkpoint_path, evaluators)
        ....
        search_algo.run()
        ....
        torch.save(evaluators.get_state(), path_to_external_evaluator)
```